test/types/valueof-test.js

```diff
@@ -0,0 +1,45 @@
+import assert from "assert";
+import {valueof} from "../../src/options.js";
+
+it("valueof reads arrays", () => {
```

Comment:
We already have tests for many of these cases in test/options-test.js We should consolidate.

We still don’t have test coverage for is the valueof(null, array) case you mentioned.